autoload/kite/codenav.vim

```diff
@@ -20,20 +20,24 @@ endfunction
 
 
 function! kite#codenav#handler(response) abort
-  if a:response.status != 200
-    if a:response.status == 0
-      call kite#utils#warn("Kite could not be reached. Please check that Kite Engine is running.")
-      return
-    endif
+  if a:response.status == 200 | return | endif
 
+  if a:response.status == 0
+    call kite#utils#warn("Kite could not be reached. Please check that Kite Engine is running.")
+    return
+  endif
 
-    let err = json_decode(a:response.body)
+  if a:response.status == 503
```

Comment:
There's another 503 that can come from this endpoint (when code finder is still indexing). I think we'll need to use something included in the response to distinguish. Perhaps an additional json property like this https://github.com/kiteco/kiteco/pull/12388/files would work if the notification body text on its own
> Kite Free gives you %d completions per day to use. Upgrade to Kite Pro to Unlock Kite now and code faster with machine learning powered completions.

is too long here or otherwise unsuitable / unclear? cc @its-dhung if you have an opinion for what the copy of the vim warning should be

Reply:
That's too long for warning displayed in vim's command line area.  I'd suggest 70 characters or fewer (the fewer the better, really).

Reply:
The text in the linked PR would work well.

Reply:
Yeah it should be. I'll merge it so you can test it with a dev build on master

Reply:
I removed the special case for 503 responses.  Now the `message` from the server will be used, if there is one.

Reply:
Thanks, this looks great